### PR TITLE
add presets for range mode

### PR DESCRIPTION
**Pull request by depyronick**
i needed it urgently with one of my projects. so i implemented it on fork, and will use it with my own npm package until we're agreed with the pull request. it seems like you're a bit insistant about formatting. i really don't have time to check them. just configure editorconfig and tslint accordingly, so none bother with it. here it goes.

### CODE REVIEW

**Comment by maeqlin** [comment]
LGTM

**Comment by CodeMaster34** [comment]
very good. it worked

File: saturn-datepicker/src/datepicker/datepicker.ts [hunk 1]
```diff
@@ -130,6 +135,66 @@ export class SatDatepickerContent<D> extends _SatDatepickerContentMixinBase
...
+
+    this._calendar.activeDate = this._calendar.beginDate;
+    this._calendar.beginDateSelectedChange.emit(this._calendar.beginDate);
+    this._calendar.dateRangesChange.emit({ begin: this._calendar.beginDate, end: this._calendar.endDate });
+
+    // close with delay, just to show user the selected range
+    setTimeout(() => {
+      this.datepicker.close();
+    }, 200);
+  }
```
Comment (SaturnTeam) [inline saturn-datepicker/src/datepicker/datepicker.ts]: I'll remove setTimeout

Comment (depyronick) [reply saturn-datepicker/src/datepicker/datepicker.ts]: yeah, at first it was a good idea but meh, not anymore.

File: saturn-datepicker/src/datepicker/datepicker.ts [hunk 2]
```diff
@@ -130,6 +135,63 @@ export class SatDatepickerContent<D> extends _SatDatepickerContentMixinBase
...
+        const startOfMonth = moment().startOf('month').toDate();
+        const endOfMonth = moment().endOf('month').toDate();
+
+        this._calendar.beginDate = this.dateAdapter.deserialize(startOfMonth);
+        this._calendar.endDate = this.dateAdapter.deserialize(endOfMonth);
+        break;
+      case 'last_month':
+        const startOfPreviousMonth = moment().subtract('1', 'month').startOf('month').toDate();
+        const endOfPreviousMonth = moment().subtract('1', 'month').endOf('month').toDate();
+
```
Comment (SaturnTeam) [inline saturn-datepicker/src/datepicker/datepicker.ts]: I think you need to move this code into separate date adapter.  in this repo

File: saturn-datepicker/src/datepicker/datepicker.ts [hunk 3]
```diff
@@ -42,22 +42,24 @@ import {
...
+import { merge, Subject, Subscription } from 'rxjs';
+import { filter, take } from 'rxjs/operators';
+import { SatCalendar } from './calendar';
+import { matDatepickerAnimations } from './datepicker-animations';
+import { createMissingDateImplError } from './datepicker-errors';
+import { SatCalendarCellCssClasses } from './calendar-body';
+import { SatDatepickerInput, SatDatepickerRangeValue } from './datepicker-input';
+import { DateAdapter } from '../datetime/date-adapter';
+
+const moment = require('moment');
```
Comment (SaturnTeam) [inline saturn-datepicker/src/datepicker/datepicker.ts]: Nope! No any dependences from moment.js!

File: saturn-datepicker/src/datepicker/datepicker-content.html [hunk 4]
```diff
@@ -1,4 +1,15 @@
...
+  <div class="preset-buttons" *ngIf="datepicker.showPresets">
+      <h3 class="preset-title">Presets</h3>
+      <button mat-button (click)="setPreset('today')">Today</button>
+      <button mat-button (click)="setPreset('yesterday')">Yesterday</button>
+      <button mat-button (click)="setPreset('two_days')">2 days ago</button>
+      <button mat-button (click)="setPreset('this_week')">This week</button>
+      <button mat-button (click)="setPreset('last_week')">Last week</button>
+      <button mat-button (click)="setPreset('this_month')">This month</button>
+      <button mat-button (click)="setPreset('last_month')">Last month</button>
+  </div>
```
Comment (SaturnTeam) [inline saturn-datepicker/src/datepicker/datepicker-content.html]: This code is suitable for custom header component, don't you think?

File: saturn-datepicker/src/datepicker/datepicker.ts (hunk 3 above)
Comment (depyronick) [reply saturn-datepicker/src/datepicker/datepicker.ts]: i did not added moment.js as a new dependency, it was just there. i believe it comes from material datepicker itself.

Comment (SaturnTeam) [reply saturn-datepicker/src/datepicker/datepicker.ts]: No, it is not. I have a moment in repo (don't remember why), but moment is not required for datepicker. And I think it shouldn't

Comment (depyronick) [reply saturn-datepicker/src/datepicker/datepicker.ts]: ok i'll update.

**Comment by SaturnTeam** [comment]
Formatting issues. Again

**Comment by depyronick** [comment]
> Formatting issues. Again

close this pr, i'll open another later.

**Pull request closed without merging**